aten/src/ATen/StorageImpl.cpp

```diff
@@ -4,27 +4,28 @@
 namespace at {
 
 StorageImpl::StorageImpl(
-    at::ScalarType scalar_type,
+    at::DataType data_type,
     ptrdiff_t size,
     at::DataPtr data_ptr,
     at::Allocator* allocator,
     bool resizable)
-    : scalar_type_(scalar_type),
+    : data_type_(data_type),
       data_ptr_(std::move(data_ptr)),
       size_(size),
       resizable_(resizable),
       allocator_(allocator),
       finalizer_(nullptr) {}
 
 StorageImpl::StorageImpl(
-    at::ScalarType scalar_type,
+    at::DataType data_type,
     ptrdiff_t size,
     at::Allocator* allocator,
     bool resizable)
     : StorageImpl(
-          scalar_type,
+          data_type,
           size,
-          allocator->allocate(at::elementSize(scalar_type) * size),
+          allocator->allocate(
+              at::elementSize(dataTypeToScalarType(data_type)) * size),
```

Comment:
Once we have a TypeMeta registry, this conversion is not necessary; you should go DataType -> TypeMeta -> size, no use of ScalarType.